xbmc/powermanagement/PowerManager.cpp

```diff
@@ -327,3 +331,30 @@ void CPowerManager::SettingOptionsShutdownStatesFiller(const CSetting *setting,
 #endif
   }
 }
+
+void CPowerManager::WaitForNet() const
+{
+  CNetwork& net = g_application.getNetwork();
+
+  // check if we have at least one network interface to wait for
+  if (!net.IsAvailable())
+    return;
+
+  CLog::Log(LOGNOTICE, "%s: Waiting for a network interface to come up", __FUNCTION__);
+
+  const static int numMaxTries = 50;
```

Comment:
those are kind of ugly. I would make them configurable as max waiting time. a value of zero disables this feature.